From 67654956c27c159e29701a2b9fd9430d372a9f91 Mon Sep 17 00:00:00 2001
From: Pavel Valena <pvalena@redhat.com>
Date: Thu, 8 Aug 2024 00:21:12 +0200
Subject: [PATCH 04/13] fix(resume): always include the resume module

as we can't determine with certainity that it won't be needed.

rhel-only

Resolves: RHEL-53350

(commit messages from squashed commits:)

fix(resume): do not include resume if swap is on netdevice

Additional fix, restoring previous behavior identical to RHEL-9.

rhel-only

Resolves: RHEL-53350
---
 modules.d/95resume/module-setup.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules.d/95resume/module-setup.sh b/modules.d/95resume/module-setup.sh
index d419566e..785f681a 100755
--- a/modules.d/95resume/module-setup.sh
+++ b/modules.d/95resume/module-setup.sh
@@ -1,7 +1,9 @@
 #!/bin/bash
 
 # called by dracut
+# shellcheck disable=SC2317
 check() {
+
     swap_on_netdevice() {
         local _dev
         for _dev in "${swap_devs[@]}"; do
@@ -18,6 +20,9 @@ check() {
             # hibernation support requested on kernel command line
             return 0
         else
+            # always include resume module when not on netdevice
+            return 0
+
             # resume= not set on kernel command line
             if [[ -f /sys/power/resume ]]; then
                 if [[ "$(< /sys/power/resume)" == "0:0" ]]; then
-- 
2.47.1

